### ISSUE TITLE
Potential missing use case for aria-posinset and aria-setsize being allowed on <tr /> or role="row"

### ISSUE THREAD

**Issue by SiTaggart** [w3c/html#1066]
According to [3.2.7.4. Allowed ARIA roles, states and properties](https://www.w3.org/TR/html/dom.html#allowed-aria-roles-states-and-properties) `aria-posinset` and `aria-setsize` are reserved for very few roles within the ARIA spec.

One missing allowed use case is placing those on an element with a role of `row`.  This seems limiting for a couple of reasons:

Firstly in large, complex web apps, large data grids with thousands of rows may benefit from only ever rendering a subset of those rows for performance reasons. In those situations, you may still wish to inform a screen reader user of their _real_ position with the grid, regardless of the number of rows we've decided to render. Being able to add `aria-posinset` and `aria-setsize` to the table rows would solve for this.

Another use case, which seems slightly more conflicting, is when you're building a `treegrid`. It's conflicting in that, if I'm building a `tree`, `treeitem`s are allows to have positions and set sizes, but within ARIA, `treegrid`s don't have `treeitems`, they have child rows. So it seems logical that I should be able to apply most, if not all of the same attributes to a `row` as I can to a `treeitem`, especially when placed inside of a `treegrid`.

Thoughts?

**Comment by LJWatson** [comment]
Neither the [aria-posinset](https://www.w3.org/TR/wai-aria-1.1/#aria-posinset) or [aria-setsize](https://www.w3.org/TR/wai-aria-1.1/#aria-sizeset) can be used on an element with `role="row`, so we're unable to change this information in the HTML spec.

The attribute definitions would need to be changed in the [ARIA spec](https://www.w3.org/TR/wai-aria-1.1/#aria-). ARIA 1.1 has just gone to Proposed Recommendation (PR) and so the definitions can't be changed in that version, but raising an issue on the [ARIA Github repo](https://github.com/w3c/aria) may mean they could be changed in ARIA 2.0.
